Lab2 - Command

Przydal opened the pull request.

Przydal committed:
fix inccorect drawing and location

Przydal committed:
Merge pull request #1 from Przydal/bugFix_draw

fix inccorect drawing and location

Przydal committed:
"rename Adapter, hide second window"

Przydal committed:
remove unused second window

Przydal committed:
Merge pull request #2 from Przydal/3.1

"rename Adapter, hide second window"

Przydal committed:
implement SpecialLineDrawerAdapter, add special line driver

Przydal committed:
git commit -m "implement DriverCommand interface and implementations"

Przydal committed:
implement figure builders and factories

Przydal committed:
add not versioned files

kowallus commented on src/main/java/edu/kis/powp/jobs2d/builders/Figure1Builder.java:
```diff
@@ -0,0 +1,33 @@
+    @Override
+    public Object build() {
+        if (this.commandList.size() == 0) {
+            FiguresJoe.figureScript1(this);
```
Actually, it would be very easy to convert those Builders to a single generic functionality (to cover building command from any script). Actually it is even possible now, because we can externally execute any set of commands on Figures1Builder (but the name doesn't suggest such possibility).

kowallus commented on src/main/java/edu/kis/powp/jobs2d/builders/Figure2Builder.java:
```diff
@@ -0,0 +1,33 @@
+import java.util.List;
+
+public class Figures2Builder implements Job2dDriver, Builder {
+    private final List<DriverCommand> commandList = new ArrayList<>();
+
+    @Override
+    public void setPosition(int x, int y) {
+        this.commandList.add(new SetPositionCommand(x, y));
+    }
+
+    @Override
+    public void operateTo(int x, int y) {
+        this.commandList.add(new OperateToCommand(x, y));
+    }
+
```
Code duplication (Figures1Builder is very similar)

Przydal replied:
Removed redundant Figure2Builder.java

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/command/ComplexCommand.java:
```diff
@@ -0,0 +1,25 @@
+import java.util.List;
+
+public class ComplexCommand implements DriverCommand {
+    List<DriverCommand> commandList;
```
should be private!

Przydal replied:
added private access identifier to complexCommand list.

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/factory/FiguresFactory.java:
```diff
@@ -0,0 +1,75 @@
+        TRIANGLE
+    }
+
+    private static class Rectangle {
+        public static final int x1 = 200;
+        public static final int y1 = 200;
+
+        public static final int x2 = 200;
+        public static final int y2 = -200;
+
+        public static final int x3 = -200;
+        public static final int y3 = -200;
+
+        public static final int x4 = -200;
+        public static final int y4 = 200;
```
Plagiarism! :(

Przydal replied:
Expanded FiguresFactory to be more generic. 
Available configurable shapes: square, triangle, trapezium, cross, arrowTip.
Added multiple tests to TestJobs2dPatterns -> some shapes in multiple versions to present flexibility of the implementation.

kowallus reviewed (changes requested):
This is evident plagiarism: compare with https://github.com/iis-powp-2021z/powp_jobs2d/pull/33.
I do not accept extra tasks an I want you to rewrite factory solution to more simple and compact one, but allowing more (and more flexible) shapes.

Przydal committed:
expand FiguresFactory, remove redundant Figure2Builder, add private access specifier to ComplexCommand list

Przydal commented:
Expanded FiguresFactory to be more generic. Available configurable shapes: square, triangle, trapezium, cross, arrowTip.
Added multiple tests to TestJobs2dPatterns -> some shapes in multiple versions to present flexibility of the implementation. 
Removed redundant Figures2Builder and adjusted Figures1Builder to be less confusing with naming.
Added private access modifier to complexCommand list in ComplexCommand.

kowallus commented:
3 pts
